src/ast/nodes/ConditionalExpression.ts

```diff
@@ -68,38 +62,40 @@ export default class ConditionalExpression extends NodeBase {
 		);
 	}
 
-	initialiseChildren(parentScope: Scope) {
-		super.initialiseChildren(parentScope);
-		if (this.module.graph.treeshake) {
-			this.testValue = this.test.getValue();
+	includeInBundle() {
+		let addedNewNodes = !this.included;
+		this.included = true;
+		const testValue = this.test.getValue();
+		if (testValue === UNKNOWN_VALUE || this.test.shouldBeIncluded()) {
+			if (this.test.includeInBundle()) addedNewNodes = true;
+			if (this.consequent.includeInBundle()) addedNewNodes = true;
+			if (this.alternate.includeInBundle()) addedNewNodes = true;
+		} else if (testValue ? this.consequent.includeInBundle() : this.alternate.includeInBundle())
+			addedNewNodes = true;
+		return addedNewNodes;
+	}
 
-			if (this.testValue === UNKNOWN_VALUE) {
-				return;
-			} else if (this.testValue) {
-				this.alternate = null;
-			} else if (this.alternate) {
-				this.consequent = null;
-			}
-		}
+	reassignPath(path: ObjectPath, options: ExecutionPathOptions) {
+		if (path.length > 0) this.forEachRelevantBranch(node => node.reassignPath(path, options));
 	}
 
-	render(code: MagicString, options: RenderOptions) {
-		if (!this.module.graph.treeshake) {
+	render(
+		code: MagicString,
+		options: RenderOptions,
+		{ hasBecomeCallee, hasBecomeStatement }: NodeRenderOptions = BLANK
```

Comment:
I guess some of this could be handled via a parameter `renderedParent`. Might also come in handy once we start inlining expressions. Will experiment a little with this.